app/src/main/java/com/epfl/drawyourpath/database/FirebaseDatabase.kt

```diff
@@ -463,6 +488,86 @@ class FirebaseDatabase : Database() {
         return Utils.failedFuture(Error("Not implemented"))
     }
 
+    override fun getTournamentUID(): String? {
+        return database.child(FirebaseKeys.TOURNAMENTS_ROOT).push().key
```

Comment:
Doesn't it make a server request? . If we want to generate a collision resistant id locally, we should use 
```kotlin
UUID.randomUUID().toString()
```

Reply:
It works perfectly offline so I don't think it does. Morevoer, every examples I've seen use push() (with no argument) to get a unique ID. If in the future we realize that it actually creates some problems we could easily change it though.

Reply:
If it doesn't then your solution is even better. We can keep it like that.